APP/Components/Search/Search/Search.js

```diff
@@ -0,0 +1,103 @@
+import React, {useState,useEffect} from "react";  
+import { SafeAreaView ,StyleSheet, View,Text,TextInput, TouchableOpacity,Button} from 'react-native';
+import ItemView from "../../ItemView";
+import SearchText from "../SearchText";
+import movielist from "../../Movielist";
+import { Provider } from 'react-redux';
+import store from "../store";
+import { useNavigation } from '@react-navigation/native';
+
+
+import { useDispatch } from 'react-redux'
+
+
+const Search=() => {
+
+  const navigation = useNavigation();
+
+  const [filterdData,setfilterdData]= useState([]);
+  const [masterData,setmasterData]= useState([]);
+  const [search,setsearch]= useState('')
+
+  useEffect(() =>{
+   fetchPosts();
+    return()=>{
+    }
+  },[])
+
+
+  const fetchPosts =() =>{
+    setfilterdData(movielist);
+    setmasterData(movielist);
+  }
+
+  const searchFilter =(text) =>{
+    if(text)
+    {
+        const newData = masterData.filter((item)=>{
+        const itemData= item.title ? item.title.toUpperCase() : ''.toUpperCase();
+        const textdata= text.toUpperCase();
+        return itemData.indexOf(textdata) > -1;
+      });
+      setfilterdData(newData);
+      setsearch(text);
+      SearchText.text=text;
+      dispatch({type:'CHANGE_TEXT'})
```

Comment:
Your are dispatchign only actions here and still storing everythign in SearchText component. which is a terrible idea. You are not even making use of redux here. You have to store the valuein redux